x-pack/plugins/spaces/public/plugin.tsx

```diff
@@ -61,11 +61,14 @@ export class SpacesPlugin implements Plugin<SpacesPluginSetup, SpacesPluginStart
       getActiveSpace: () => this.spacesManager.getActiveSpace(),
     };
 
-    if (plugins.home) {
+    if (plugins.home && this.initializerContext.env.packageInfo.buildFlavor !== 'serverless') {
       plugins.home.featureCatalogue.register(createSpacesFeatureCatalogueEntry());
     }
 
-    if (plugins.management) {
+    if (
+      plugins.management &&
+      this.initializerContext.env.packageInfo.buildFlavor !== 'serverless'
+    ) {
```

Comment:
This disables the Spaces management service but I _think_ the better way would be to disable it inside the service itself. Wdyt @thomheymann @azasypkin ?

Reply:
> but I think the better way would be to disable it inside the service itself.

Would you mind elaborating on this? I was thinking that having these if/else branches in a single (top-most) place would make it easier to track serverless-specific changes, but I don't have a strong opinion and am open to any other better options!

Reply:
@azasypkin I agree with you. I've made a small change to the flag itself and i think it's much better this way than controlling it inside the management app.